### PR TITLE
Support for Vapor 4

### PR DESCRIPTION
**Pull request by PoshAlpaca**
Adds support for Vapor 4. Originally implemented by rabc. I've combined it with our custom `filterGeographyDistanceWithin` and `sortByDistance` functions.

### CODE REVIEW

**Comment by PoshAlpaca** [comment]
@0xTim  ready for review 🎉

File: Package.swift
Comment (0xTim) [inline Package.swift:1]: Let's just set this to 5.5 to make life easier

File: Package.swift
Comment (0xTim) [inline Package.swift:7]: Let's do v12 to allow us to add async/await stuff easily if we want to

File: README.md
Comment (0xTim) [inline README.md:13]: Let's set this for the Broken Hands fork set up

File: README.md
Comment (0xTim) [inline README.md:6]: Let's add a sentence about it being a fork of the PostGIS package that provides geographic support on top of the geometric support from the upstream repo

File: README.md
Comment (0xTim) [inline README.md:33]: We might as well make this a UUID as that's the default in Fluent

File: README.md
Comment (0xTim) [inline README.md:46]: Again UUID like the model

File: README.md
Comment (0xTim) [inline README.md:70]: We define a search location but don't actually use it

File: README.md
Comment (0xTim) [inline README.md:69]: Let's use a better lat/lon for this, something like Buckingham Palace etc

File: Sources/FluentPostGIS/Queries/QueryBuilder+Contains.swift
Comment (0xTim) [inline Sources/FluentPostGIS/Queries/QueryBuilder+Contains.swift:24]: Indentation seems off here in the diff

File: Sources/FluentPostGIS/Support/EnablePostGISMigration.swift
Comment (0xTim) [inline Sources/FluentPostGIS/Support/EnablePostGISMigration.swift:6]: Just double check the name of the migration is computed to be the same in Vapor 3 and Vapor 4 to make sure we don't try to run this again

File: Tests/FluentPostGISTests/GeometryTests.swift
Comment (0xTim) [inline Tests/FluentPostGISTests/GeometryTests.swift:13]: If async/await is available to us here we should use it

File: Tests/FluentPostGISTests/GeometryTests.swift
Comment (0xTim) [inline Tests/FluentPostGISTests/GeometryTests.swift:51]: a/a if possible

File: Tests/FluentPostGISTests/GeometryTests.swift
Comment (0xTim) [inline Tests/FluentPostGISTests/GeometryTests.swift:51]: `save()` returns a `Void` so the `_` can be removed

File: Tests/FluentPostGISTests/XCTestManifests.swift
Comment (0xTim) [inline Tests/FluentPostGISTests/XCTestManifests.swift:5]: Let's delete this file and rely on test discovery instead (assuming it works with the subclassing!)

**Review by 0xTim — CHANGES_REQUESTED** [review]
Great work! Few comments and things that can be updated. Some other things to look at are:
* use async/await throughout the tests assuming it's available to us. Updating the manifest to 5.5 and macOS 12 will help
* we don't have any CI any more, we should set up GH actions to run the tests to make sure things work

**Comment by PoshAlpaca** [comment]
@0xTim Thanks for the feedback! Can you take a look again?

**Comment by PoshAlpaca** [comment]
Also, I was thinking to extend `DatabaseSchema.DataType` with static properties so that fields in migrations can be declared like so:

```swift
.field("location", .geometricPoint2D)
```

instead of:

```swift
.field("location", GeometricPoint2D.dataType)
```

What do you think?

File: .github/workflows/ci.yml
Comment (0xTim) [inline .github/workflows/ci.yml:3]: We should add pull request here as well

File: .github/workflows/ci.yml
Comment (0xTim) [inline .github/workflows/ci.yml:19]: Don't need test discovery that's default since 5.4

**Comment by 0xTim** [comment]
@PoshAlpaca yep I think that's a great idea to provide a cleaner API

**Review by 0xTim — CHANGES_REQUESTED** [review]
Looking great! Last couple of changes just to CI then we're good. We should have CI passing in this PR

And let's add the data type stuff as well if it's simple

**Review by 0xTim — APPROVED** [review]
Looks great! Might need to tweak the `on` section of CI to get it to run or wait until it's merged. Happy to merge however and fix in another PR if necessary

**Pull request merged**
